### PR TITLE
feat(chart): allow to customize the storageclass

### PR DESCRIPTION
**Pull request by synthe102**
This pull request aims to allow the customization of the `storageClass` created by the mayastor helm chart.

## Description

The following capabilities are added for the `storageClass`:
- Allow to disable the creation
- Allow to set the `storageClass` as the default one
- Allow to set the replication factor
- Allow to set the name

## Motivation and Context

The main motivation for those changes is to make the `storageClass` configurable and usable as default in a k8s cluster without the need to create a second one from a separate set of manifests.

## Regression

No

## How Has This Been Tested?

Tested by deploying Mayastor into a kubernetes cluster and validating that the template is rendered as valid k8s yaml.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -705,6 +705,12 @@ obs:
         # NodePort associated with https port
         https: 90010
 
+storageClass:
+  enabled: true
+  nameSuffix: single-replica
+  default: false
+  repl: 1
```
Comment (tiagolobocastro) [inline chart/values.yaml:712]: Wonder if it'd be useful to have this under a parameters key?
CC @niladrih  @Abhinandan-Purkait

Comment (synthe102) [reply chart/values.yaml:712]: Hi ! It could help with keeping the values tidy.
I can fix this if you want, just let me know.

Comment (tiagolobocastro) [reply chart/values.yaml:712]: I just chatted to @Abhinandan-Purkait, yes please place parameters inside parameters key. Thank you!

Comment (synthe102) [reply chart/values.yaml]: Done ✅

File: chart/values.yaml [hunk 2]
```diff
@@ -705,6 +705,13 @@ obs:
         # NodePort associated with https port
         https: 90010
 
+storageClass:
+  enabled: true
+  parameters:
+    nameSuffix: single-replica
```
Comment (tiagolobocastro) [inline chart/values.yaml:711]: this probably leave it on StorageClass as it's not a parameter?

Comment (synthe102) [reply chart/values.yaml]: Makes sense, fixed.

**Review by tiagolobocastro — APPROVED** [review]
@synthe102 could you please squash that merge commit?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: chart/values.yaml [hunk 3]
```diff
@@ -705,6 +705,13 @@ obs:
         # NodePort associated with https port
         https: 90010
 
+storageClass:
+  enabled: true
+  nameSuffix: single-replica
+  parameters:
+    default: false
```
Comment (Abhinandan-Purkait) [inline chart/values.yaml:712]: @tiagolobocastro @synthe102 Should this also not be a part of the parameters?

Comment (tiagolobocastro) [reply chart/values.yaml:712]: oh I missed that, yes that should also not be on parameters

**Review by tiagolobocastro — CHANGES_REQUESTED** [review]

File: chart/values.yaml (hunk 3 above)
Comment (tiagolobocastro) [reply chart/values.yaml:712]: Could you also move this @synthe102 ? Thanks

**Review by tiagolobocastro — APPROVED** [review]
:rocket:

**Comment by tiagolobocastro** [comment]
Thanks @synthe102 :+1: 
bors merge

**Pull request merged**